soc/st/stm32/common/Kconfig.defconfig

```diff
@@ -0,0 +1,9 @@
+if SOC_SERIES_STM32N6X && DT_HAS_ST_STM32_IOCELL_ENABLED
```

Comment:
If SoC specific, then it should be moved to soc/stm32n6xx. Keep it here if not functional otherwise (and then please add header).